code/modules/surgery/bodyparts/head_hair_and_lips.dm

```diff
@@ -1,121 +1,362 @@
 			if(hair_gradient_style)
-				hair_gradient_color = LAZYACCESS(human_head_owner.grad_color, GRADIENT_HAIR_KEY)
-				hair_gradient_overlay = make_gradient_overlay(sprite_accessory.icon, sprite_accessory.icon_state, HAIR_LAYER, GLOB.hair_gradients_list[hair_gradient_style], hair_gradient_color)
-			//Emissive
-			hair_overlay.overlays += emissive_blocker(hair_overlay.icon, hair_overlay.icon_state, location, alpha = hair_alpha)
+				var/hair_gradient_color = LAZYACCESS(gradient_colors, GRADIENT_HAIR_KEY)
+				var/image/hair_gradient_overlay = get_gradient_overlay(sprite_accessory.icon, sprite_accessory.icon_state, HAIR_LAYER, GLOB.hair_gradients_list[hair_gradient_style], hair_gradient_color)
+				. += hair_gradient_overlay
+			//Emissive blocker
+			if(blocks_emissive)
+				hair_overlay.overlays += emissive_blocker(hair_overlay.icon, hair_overlay.icon_state, location, alpha = hair_alpha)
+			//Offsets
+			worn_face_offset?.apply_offset(hair_overlay)
+			. += hair_overlay
+
+	if(show_debrained && (head_flags & HEAD_DEBRAIN))
+		. += get_debrain_overlay(can_rotate = !dropped)
+
+	if(show_eyeless && (head_flags & HEAD_EYEHOLES))
+		. += get_eyeless_overlay(can_rotate = !dropped)
 
 	//HAIR COLOR START
-	if(!override_hair_color)
-		if(hair_color_source)
-			if(hair_color_source == "fixedmutcolor")
-				SET_OVERLAY_VALUE(facial_overlay, color, fixed_hair_color)
-				SET_OVERLAY_VALUE(hair_overlay, color, fixed_hair_color)
-			else if(hair_color_source == "mutcolor")
-				SET_OVERLAY_VALUE(facial_overlay, color, facial_hair_color)
-				SET_OVERLAY_VALUE(hair_overlay, color, hair_color)
-			else
-				SET_OVERLAY_VALUE(facial_overlay, color, hair_color_source)
-				SET_OVERLAY_VALUE(hair_overlay, color, hair_color_source)
-		else
-			SET_OVERLAY_VALUE(facial_overlay, color, facial_hair_color)
-			SET_OVERLAY_VALUE(hair_overlay, color, hair_color)
-	else
-		SET_OVERLAY_VALUE(facial_overlay, color, override_hair_color)
+	if(override_hair_color)
+		SET_OVERLAY_VALUE(facial_hair_overlay, color, override_hair_color)
 		SET_OVERLAY_VALUE(hair_overlay, color, override_hair_color)
+	else
+		SET_OVERLAY_VALUE(facial_hair_overlay, color, facial_hair_color)
+		SET_OVERLAY_VALUE(hair_overlay, color, hair_color)
 	//HAIR COLOR END
 
+	return .
+
 #undef SET_OVERLAY_VALUE
+
+/// Returns an appropriate debrained overlay
+/obj/item/bodypart/head/proc/get_debrain_overlay(can_rotate = TRUE)
+	RETURN_TYPE(/image)
+	var/debrain_icon = 'icons/mob/species/human/human_face.dmi'
+	var/debrain_icon_state = "debrained"
+	if(bodytype & BODYTYPE_ALIEN)
+		debrain_icon = 'icons/mob/species/alien/bodyparts.dmi'
+		debrain_icon_state = "debrained_alien"
+	else if(bodytype & BODYTYPE_LARVA_PLACEHOLDER)
+		debrain_icon = 'icons/mob/species/alien/bodyparts.dmi'
+		debrain_icon_state = "debrained_larva"
+	else if(bodytype & BODYTYPE_GOLEM)
+		debrain_icon = 'icons/mob/species/golems.dmi'
+		debrain_icon_state = "debrained"
```

Comment:
In a perfect world these would just be vars on the head. It's fine for now though

Reply:
Soon (TM)